lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -37,40 +37,37 @@ def << arg
     end
 
     def collections= collections
-      # check that each collection is an instance of Hydra::PCDM::Collection
-      raise ArgumentError, "each collection must be a Hydra::PCDM::Collection" unless
-          collections.all? { |c| c.is_a? Hydra::PCDM::Collection }
-
-      # TODO - how to prevent A - B - C - A causing a recursive loop of collections?
-
-      current_objects = self.objects
-      new_members = current_objects + collections
-      self.members = new_members
+      raise ArgumentError, "each collection must be a Hydra::PCDM::Collection" unless collections.all? { |c| collection? c }
```

Comment:
Is it safe to assume that everything in collections will respond to `collection?`  Seems like anything that's not a collection will raise a MethodNotDefined error.